specs/agents/universal-profiling-integration.md

```diff
@@ -173,6 +171,23 @@ All messages have the following layout:
 * *message-type* : An ID uniquely identifying the type (and therefore payload structure) of the message.
 * *minor-version* : The version number for the given *message-type*. This value is incremented when new fields are added to the payload while preserving the *message-type* (non breaking changes). For breaking changes a new *message-type* must be used.
 
+## Profiler Registration Message
+
+Whenever the profiling host agent starts communicating for the first time with a process running an APM Agent, it MUST send this message.
+This message is used to let the APM-agent know that a profiler is actually active on the current host. Note that that an APM-agent may receive this message zero, one or several times: This may happen if no host agent is active, if one is active or if a host agent is restarted during the lifetime of the APM-agent respectively.
```

Comment:
```suggestion
This message is used to let the APM-agent know that a profiler is actually active on the current host. Note that an APM-agent may receive this message zero, one or several times: This may happen if no host agent is active, if one is active or if a host agent is restarted during the lifetime of the APM-agent respectively.
```